---
title: "Untitled"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(readr)
library(magrittr)
library(dplyr)
library(here)
```

Identify GTEx variant ID corresponding to each SNP in the library
```{r}
lib <- read_tsv(here("output/lib_studies_idmerge_diseasesexpanded_coords_gtex_hichip_gwasrisk_1.14.20.tsv"))

gtex_lookup <- read_tsv(here("data/GTEx_Analysis_2017-06-05_v8_WholeGenomeSeq_838Indiv_Analysis_Freeze.lookup_table.txt"))


```


```{r}
all_snps = unique(c(lib$Causal_SNP, lib$Lead_SNP))

gtex_rsids <- gtex_lookup %>% filter(rs_id_dbSNP151_GRCh38p7 %in% all_snps)
gtex_leftover <- all_snps[!(all_snps %in% gtex_rsids$rs_id_dbSNP151_GRCh38p7)]

# 426 total leftover
# 341 causal snps
```

426 leftovers. 
I checked if their merged ids are in there, but no.


```{r}
gtex_rsids_causal <- dplyr::filter(gtex_rsids, rs_id_dbSNP151_GRCh38p7 %in% lib$Causal_SNP) %>%
  dplyr::select(variant_id, rs_id_dbSNP151_GRCh38p7, num_alt_per_site, alt) 

# need to deal with the duplicates in a way that won't interfere later with merging in e-genes.
# Maybe allow multiple entries

lib <- left_join(lib, gtex_rsids_causal,
                 by=c("Causal_SNP" = "rs_id_dbSNP151_GRCh38p7"))
# expands it by a few
write_tsv(lib, here("output/lib_studies_idmerge_diseasesexpanded_coords3738_gtexIDuncollapsed_1.9.20.tsv")) 

```