---
title: "Untitled"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(readr)
```

```{r}
lib <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/AnalysisFiles12.5.19/FormattingBackground/GenomicCoordinates/lib_studies_idmerge_diseasesexpanded_coords3738_12.13.19.tsv")


gtex_lookup <- read_tsv("~/Documents/MPRA/Cancer_MPRA/Lx_Analysis/GTEx_Analysis_v8_eQTL/GTEx_Analysis_2017-06-05_v8_WholeGenomeSeq_838Indiv_Analysis_Freeze.lookup_table.txt")


```


```{r}
all_snps = unique(c(lib$Causal_SNP, lib$Lead_SNP))

gtex_rsids <- gtex_lookup %>% filter(rs_id_dbSNP151_GRCh38p7 %in% all_snps)
gtex_leftover <- all_snps[!(all_snps %in% gtex_rsids$rs_id_dbSNP151_GRCh38p7)]

# 426 total leftover
# 341 causal snps
```

426 leftovers. 
I checked if their merged ids are in there, but no.


```{r}
gtex_rsids_causal <- dplyr::filter(gtex_rsids, rs_id_dbSNP151_GRCh38p7 %in% lib$Causal_SNP) %>%
  dplyr::select(variant_id, rs_id_dbSNP151_GRCh38p7, num_alt_per_site, alt) 

# need to deal with the duplicates in a way that won't interfere later with merging in e-genes.
# Maybe allow multiple entries

lib <- left_join(lib, gtex_rsids_causal,
                 by=c("Causal_SNP" = "rs_id_dbSNP151_GRCh38p7"))
# expands it by a few
write_tsv(lib, "~/Documents/MPRA/Cancer_MPRA_Scaleup/AnalysisFiles12.5.19/FormattingBackground/GTEx_IDs_eQTLs/lib_studies_idmerge_diseasesexpanded_coords3738_gtexIDuncollapsed_1.9.20.tsv")
# gtex_rsids_causal_sum <- gtex_rsids_causal %>% group_by(rs_id_dbSNP151_GRCh38p7) %>%
  summarise(gtex_alt = paste(unique(alt), collapse=","), 
            gtex_num_alt_per_site = unique(num_alt_per_site),
            gtex_variant_id = paste(sort(unique(variant_id)), collapse=","))

# lib <- left_join(lib, gtex_rsids_causal_sum,
                 by=c("Causal_SNP" = "rs_id_dbSNP151_GRCh38p7"))
   

# write_tsv(lib, "~/Documents/MPRA/Cancer_MPRA_Scaleup/AnalysisFiles12.5.19/FormattingBackground/GTEx_IDs_eQTLs/lib_studies_idmerge_diseasesexpanded_coords3738_gtexID_12.13.19.tsv")
```

```{r}


```



Checking merged ids - ignore

```{r}
merge_conversion <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/AnalysisFiles12.5.19/FormattingBackground/MergingSNPs/lib_snps_merged_rsid_conversion.tsv")
merge_conversion <- merge_conversion %>% mutate(causal_merged_rsID = paste("rs", causal_merged_rsID, sep=""),
                                                lead_merged_rsID = paste("rs", lead_merged_rsID, sep=""))
merge_leftovers_causal <- dplyr::filter(merge_conversion, causal_merged_rsID %in% gtex_leftover)
merge_leftovers_lead <- dplyr::filter(merge_conversion, lead_merged_rsID %in% gtex_leftover)
sum(!(merge_leftovers_causal$causal_merged_rsID == merge_leftovers_causal$Causal_SNP))
# 17
sum(!(merge_leftovers_lead$lead_merged_rsID == merge_leftovers_lead$leadSNP))
# 0

causal_merge_changes = merge_leftovers_causal[!(merge_leftovers_causal$causal_merged_rsID == merge_leftovers_causal$Causal_SNP),]
sum(causal_merge_changes$Causal_SNP %in% gtex_lookup$rs_id_dbSNP151_GRCh38p7)
# so looks like merge won't help. 
```



