---
title: "Untitled"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(readr)
library(tidyverse)
library(viridis)
```

Read 
```{r}
eqtlgen <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/eGeneFiles/eQTLgen/2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt")
```

```{r}
res_merge <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/AnalysisFiles12.5.19/DataAnalysis/Analysis_9.21.20/res_merged_9.21.20.tsv")

res_hmec <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/Sequencingv3/NextSeq_Pool1/Temp_HMEC_Analysis/mpranalyze_hmecNextSeq_oldpGF.tsv")
res_merge <- right_join(res_hmec, res_merge)
res_merge <- dplyr::filter(res_merge, !is.na(Causal_SNP))
```


```{r}
sum(unique(eqtlgen$SNP) %in% res_merge$Causal_SNP)
sum(unique(res_merge$Causal_SNP) %in% eqtlgen$SNP)

```


```{r}
eg_snps <- distinct(dplyr::select(eqtlgen, SNP, SNPChr, SNPPos, AssessedAllele, OtherAllele))
# how many snps would be gained by using the position instead of rsid?
res_snps <- distinct(dplyr::select(res_merge, Causal_SNP, Chr_37, Start_37, Ref, Alt))

check_pos <- left_join(res_snps, eg_snps, by=c("Chr_37"="SNPChr", "Start_37"="SNPPos"))
sum(!is.na(check_pos$SNP))
# 3496
# 3828?
check_pos_id <- left_join(check_pos, dplyr::select(eg_snps, -AssessedAllele, -OtherAllele),
                          SNP, by=c("Causal_SNP"="SNP"))
sum(!is.na(check_pos_id$SNPPos))
# 3058
# 3083?
test <- check_pos_id[is.na(check_pos_id$SNPPos) & !is.na(check_pos_id$SNP),]
# using position saves some rsid inconsistencies, so add them as well
```




```{r}
snpidlist <- unique(c(check_pos$Causal_SNP, check_pos$SNP))
# filter to only eatls in our library
eqtlgen <- dplyr::filter(eqtlgen, SNP %in% snpidlist)
# add a column to eqtlgen with the rsid the lib uses
eqtlgen <- eqtlgen %>% rowwise() %>%
  mutate(SNP_ID_lib = ifelse(SNP %in% check_pos$SNP, 
                             check_pos$Causal_SNP[check_pos$SNP == SNP & !is.na(check_pos$SNP)],
                             NA))
                             
            

```

Go to check_pos, start categorizing the allele matchups
```{r}
check_alleles <- check_pos %>% rowwise() %>%
  mutate(eqtlgen_match = case_when(
    AssessedAllele == Ref & OtherAllele == Alt ~ "same",
    OtherAllele == Ref & AssessedAllele == Alt ~ "flip",
    is.na(AssessedAllele) ~ "no_eqtlgen",
    T ~ "eqtlgen_allele_mismatch"
  ))

# what information do I want on the eqtls?
# is there an eqtl? - eqtlgen_match, merge in on CausalSNP, Ref, Alt
# what are the egenes? - eqtlgen_egenes, collapse, semicolon separate
# what's the direction? - eqtlgen_dir, collapse, semicolon
# all same direction or diff? - eqtlgen_dir_sum, have same, diff


```

Check if eqtl gen ever assesses multiple alleles - it doesn't
```{r}
check_mult <- eqtlgen %>% group_by(SNP) %>% summarise(num_alleles = length(unique(AssessedAllele)))

```

```{r}
check_alleles_onlymatch <- dplyr::filter(check_alleles, eqtlgen_match == "flip" | eqtlgen_match == "same")
eqtlgen_flip <- left_join(eqtlgen, dplyr::select(check_alleles_onlymatch, Causal_SNP, eqtlgen_match),
                          by=c("SNP_ID_lib"="Causal_SNP"))
collapse_eg_flip <- eqtlgen_flip %>% 
  mutate(Zscore_flip = ifelse(eqtlgen_match=="flip", -1*Zscore, Zscore),
         Ref = ifelse(eqtlgen_match == "flip", OtherAllele, AssessedAllele),
         Alt = ifelse(eqtlgen_match == "flip", AssessedAllele, OtherAllele)) %>%
  group_by(SNP_ID_lib, AssessedAllele, OtherAllele, Ref, Alt, eqtlgen_match) %>%
  summarise(eqtlgen_egenes = paste(GeneSymbol, collapse=";"),
            eqtlgen_dir_flipped =  paste(Zscore_flip, collapse=";"))



collapse_eg <- eqtlgen %>% group_by(SNP_ID_lib, AssessedAllele, OtherAllele) %>%
  summarise(eqtlgen_egenes = paste(GeneSymbol, collapse=";"),
            eqtlgen_dir_unflipped = paste(Zscore, collapse=";"))
```
Add eqtlgen data into res_merge
```{r}
res_merge_eg <- left_join(res_merge, 
                          dplyr::select(ungroup(collapse_eg_flip), SNP_ID_lib, Ref, Alt,
                                        eqtlgen_egenes, eqtlgen_dir_flipped),
                          by=c("Causal_SNP"="SNP_ID_lib", "Ref"="Ref", "Alt"="Alt"))

# write_tsv(res_merge_eg, "~/Documents/MPRA/Cancer_MPRA_Scaleup/eGeneFiles/eQTLgen/res_merge_witheQTLgen_11.1.2020.tsv")

res_merge_eg <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/eGeneFiles/eQTLgen/res_merge_witheQTLgen_11.1.2020.tsv")
```

How many eQTLgen egenes per snp?
```{r fig.width=2.5, fig.height=1.5}
eg_sum <- res_merge_eg %>% rowwise() %>% 
  mutate(num_eqtlgen = ifelse(is.na(eqtlgen_egenes), 0, length(unlist(str_split(eqtlgen_egenes, ";")))))
eg_sum <- eg_sum %>% mutate(eqtlgen_dir_agree = case_when(
  is.na(eqtlgen_dir_flipped) ~ "NA",
  length(unlist(str_split(eqtlgen_dir_flipped, ";"))) == 1 ~ "agree",
  sum(unlist(str_split(eqtlgen_dir_flipped, ";")) < 0) > 0 &
           sum(unlist(str_split(eqtlgen_dir_flipped, ";")) > 0) > 0 ~ "mixed",
  T ~ "agree"))


eg_sum <- eg_sum %>%  mutate(eqtlgen_hmecmpra_agree = case_when(
  is.na(eqtlgen_dir_flipped) ~ "NA",
  sum(unlist(str_split(eqtlgen_dir_flipped, ";")) < 0) > 0 &
           sum(unlist(str_split(eqtlgen_dir_flipped, ";")) > 0) > 0 ~ "mixed",
  sum(unlist(str_split(eqtlgen_dir_flipped, ";")) < 0) == length(unlist(str_split(eqtlgen_dir_flipped, ";"))) &
    hmec_logFC < 0 ~ "down",
  sum(unlist(str_split(eqtlgen_dir_flipped, ";")) > 0) == length(unlist(str_split(eqtlgen_dir_flipped, ";"))) &
    hmec_logFC > 0 ~ "up",
  T ~ "disagree"))


         
eg_sum_hmechits <- eg_sum %>% dplyr::filter(hmec_fdr < 0.01 & abs(hmec_logFC) > 0.5) 


ggplot(eg_sum, aes(x = num_eqtlgen, fill =eqtlgen_dir_agree)) + geom_histogram(bins = 46) + 
  ggtitle("Number of eQTL-Gen e-genes per SNP") +
  xlab("Number of eQTL-Gen e-genes") +
  ylab("Number of SNPs") +
  labs(fill = "Do directions \nagree?") +
  scale_fill_viridis(discrete=TRUE, option ="D") +
  theme_classic() 

ggplot(eg_sum, aes(x = num_eqtlgen, fill =eqtlgen_hmecmpra_agree)) + geom_histogram(bins = 46) + 
  ggtitle("Number of eQTL-Gen e-genes per SNP") +
  xlab("Number of eQTL-Gen e-genes") +
  ylab("Number of SNPs") +
  labs(fill = "Do directions \nagree?") +
  scale_fill_viridis(discrete=TRUE, option ="D") +
  theme_classic() 

ggplot(eg_sum_hmechits, aes(x = num_eqtlgen, fill =eqtlgen_hmecmpra_agree)) + geom_histogram(bins = 46) + 
  ggtitle("Number of eQTL-Gen e-genes per SNP, \n HMEC hits only (FDR < 0.01, |logFC| > 0.5)") +
  xlab("Number of eQTL-Gen e-genes") +
  ylab("Number of SNPs") +
  labs(fill = "Does GTEx \ndirection agree \nwith HMEC MPRA?") +
  scale_fill_viridis(discrete=TRUE, option ="D") +
  theme_classic() 

```

GTEX
```{r}
# gtex_pairs <- read_tsv("~/Documents/MPRA/Cancer_MPRA/Lx_Analysis/significant_egene_pairs_for_snps_in_gtex.txt")
gtex_med <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/eGeneFiles/GTEx/Uncollapsed_tissuefilt_egenesnp_pairs.txt")
# the gtex NES is wrt the second allele they list in the variant id
# ie, if variant id is chr1_160262886_G_A_b38, in the amygdala COPA slope is .29
# violin plot AA is higher

# split variantid into alleles
gtex_vars <- gtex_med %>% dplyr::select(variant_id) %>% distinct() %>%
  separate(variant_id, into=c("gtex_chr", "gtex_pos", "gtex_other_allele", "gtex_assessed_allele", "build"), remove=F)

gtex_vars_merge <- full_join(dplyr::select(res_merge, locus_name, Causal_SNP, Ref, Alt, Disease_expanded, 
                                           variant_id, num_alt_per_site, gtex_egenes),
                             dplyr::select(gtex_vars, variant_id, gtex_other_allele, gtex_assessed_allele),
                             by=c("variant_id"= "variant_id"))
gtex_vars_merge <- dplyr::filter(gtex_vars_merge, !is.na(locus_name))

rc_base <- function(x) {
  return(case_when(
    x=="G" ~ "C",
    x=="C" ~ "G",
    x=="A" ~ "T",
    x=="T" ~ "A"))
}

gtex_vars_merge <- gtex_vars_merge %>% 
  mutate(gtex_allele_match = case_when(
    Ref == gtex_assessed_allele & Alt == gtex_other_allele ~ "match",
    Alt == gtex_assessed_allele & Ref == gtex_other_allele ~ "flip",
    is.na(gtex_egenes) ~ "no_egenes",
    # When Ref is the deletion the assessed_allele is always the nondel (alt), so flip
    # this isn't a principle, this is just from looking at the lib
    # When alt is the deletion, the assessed_allele is always the del (alt) so flip
    # if Ref is a deletion and gtex_assessed_allele is deletion, they match
    Ref == "-" | Alt == "-" ~ "flip",
    # about 293 have mismatches. How many are RCs?
    rc_base(Ref) == gtex_assessed_allele & rc_base(Alt) == gtex_other_allele ~ "RC_match",
    rc_base(Alt) == gtex_assessed_allele & rc_base(Ref) == gtex_other_allele ~ "RC_flip",
    T ~ "mismatch"))



```

```{r}
# want to deduplicate (sort of)
# then weave back in the egenes, tissues and directions
# then flip, for both all and tissue specific
gtex_vars_filt <- dplyr::filter(gtex_vars_merge, !gtex_allele_match == "no_egenes")
# leaves 383 variant ids with duplicates where variant ID 
# we end up with 2 loci that have duplicate entries, one entry with gtex egenes that match and one with
# other set of alleles (it's when there's a gtex duplicate allele that I didn't test in the library)


 test <- dplyr::filter(gtex_vars_filt, variant_id %in%
                         gtex_vars_filt$variant_id[duplicated(gtex_vars_filt$variant_id) &
                                                     !is.na(gtex_vars_filt$variant_id)])
 test2 <-  dplyr::filter(test, locus_name %in%
                         test$locus_name[duplicated(test$locus_name) &
                                                     !is.na(test$locus_name)])

 # merge in from gtex_med the slope, tissue, and hgnc
gtex_vars_data <- left_join(dplyr::select(gtex_vars_filt, -num_alt_per_site,-gtex_egenes),
                            dplyr::select(gtex_med, variant_id, tissue, slope, hgnc),
                            by="variant_id")
# there's some loss in using HGNC rather than ensembl gene names, ignoring it for now
gtex_vars_data <- dplyr::filter(gtex_vars_data, !is.na(hgnc))


```

Flip gtex alleles so directions match
```{r}
gtex_flip <- gtex_vars_data %>% 
  mutate(gtex_dir_flip = case_when(
    gtex_allele_match == "flip" ~ -1*slope,
    gtex_allele_match == "match" ~ slope,
    gtex_allele_match == "mismatch" ~ slope,
    gtex_allele_match == "RC_flip" ~ -1*slope))

```


Group and summarise. Include whether direction agrees or is mixed.
```{r fig.width=2.5, fig.height=1.5}
gtex_sum_all <- gtex_flip %>%
  group_by(locus_name, Causal_SNP, Ref, Alt, Disease_expanded, variant_id, gtex_allele_match) %>%
  summarise(gtex_egenes_all = paste(hgnc, collapse=";"),
            gtex_dir_agree = case_when(
              sum(gtex_dir_flip < 0) == length(gtex_dir_flip) ~ "agree",
              sum(gtex_dir_flip > 0) == length(gtex_dir_flip) ~ "agree",
              T ~ "mixed"),
            gtex_dir_all = paste(gtex_dir_flip, collapse=";"),
            gtex_tissue_all = paste(tissue, collapse=";"),
            gtex_egenes_num = length(unique(hgnc)))



res_merge_gtex <- left_join(dplyr::select(res_merge_eg, -num_alt_per_site, -alt, -gtex_egenes,
                                          -gtex_direction, -gtex_egenes_correct_tissue,
                                          -gtex_egenes_ens, -gtex_tissues),
                          dplyr::select(ungroup(gtex_sum_all), locus_name, variant_id,
                                        gtex_allele_match, gtex_egenes_all, gtex_tissue_all, 
                                        gtex_dir_all, gtex_dir_agree, gtex_egenes_num))
res_merge_gtex <- res_merge_gtex %>% 
  mutate(gtex_egenes_num = ifelse(is.na(gtex_egenes_num), 0, gtex_egenes_num),
         gtex_dir_agree = ifelse(is.na(gtex_dir_agree), "NA", gtex_dir_agree))

ggplot(res_merge_gtex, aes(x = gtex_egenes_num, fill =gtex_dir_agree)) + geom_histogram(bins=41) + 
  ggtitle("Number of GTEx e-genes per SNP (all tissues)") +
  xlab("Number of GTEx e-genes") +
  ylab("Number of SNPs") +
  labs(fill = "Do directions \nagree?") +
  scale_fill_viridis(discrete=TRUE, option ="D") +
  theme_classic() 

res_merge_gtex_sum <- res_merge_gtex %>%  rowwise() %>%
  mutate(gtex_hmecmpra_agree = case_when(
    gtex_egenes_num == 0 ~ "NA",
    sum(unlist(str_split(gtex_dir_all, ";")) < 0) > 0 & 
      sum(unlist(str_split(gtex_dir_all, ";")) > 0) > 0 ~ "mixed",
    sum(unlist(str_split(gtex_dir_all, ";")) < 0) == length(unlist(str_split(gtex_dir_all, ";"))) &
    hmec_logFC < 0 ~ "down",
  sum(unlist(str_split(gtex_dir_all, ";")) > 0) == length(unlist(str_split(gtex_dir_all, ";"))) &
    hmec_logFC > 0 ~ "up",
  T ~ "disagree"))  

res_merge_gtex_hmec_hits <- dplyr::filter(res_merge_gtex_sum, abs(hmec_logFC) > 0.5 & hmec_fdr < 0.01)




ggplot(res_merge_gtex_hmec_hits, aes(x = gtex_egenes_num, fill =gtex_hmecmpra_agree)) + 
  geom_histogram(bins = 39) + 
  ggtitle("Number of GTEx e-genes per SNP, \n HMEC hits only (FDR < 0.01, |logFC| > 0.5)") +
  xlab("Number of GTEx e-genes") +
  ylab("Number of SNPs") +
  labs(fill = "Does GTEx \ndirection agree \nwith HMEC MPRA?") +
  scale_fill_viridis(discrete=TRUE, option ="D") +
  theme_classic() 

 

```

Now make tissue specific
```{r fig.width=2.5, fig.height=1.5}
gtex_flip_spec <- gtex_flip %>% rowwise() %>%
  dplyr::filter(case_when(
    grepl("airway", Disease_expanded) & grepl("Lung", tissue) ~ T,
    grepl("ast", Disease_expanded) & grepl("Brain", tissue) ~ T,
    grepl("brca", Disease_expanded) & grepl("Breast", tissue) ~ T,
    grepl("colon", Disease_expanded) & grepl("Colon", tissue) ~ T,
    grepl("endo", Disease_expanded) & grepl("Uterus", tissue) ~ T,
    grepl("eso", Disease_expanded) & grepl("Esophagus", tissue) ~ T,
    grepl("lymphoma", Disease_expanded) & grepl("EBV", tissue) ~ T,
    grepl("thy", Disease_expanded) & grepl("Thyroid", tissue) ~ T,
    grepl("pros", Disease_expanded) & grepl("Prostate", tissue) ~ T,
    grepl("kc", Disease_expanded) & grepl("Skin", tissue) ~ T,
    grepl("melanoma", Disease_expanded) & grepl("Skin", tissue) ~ T,
    grepl("panc", Disease_expanded) & grepl("Pancreas", tissue) ~ T,
    grepl("renal", Disease_expanded) & grepl("Kidney", tissue) ~ T,
    grepl("ovca", Disease_expanded) & grepl("Ovary", tissue) ~ T,
    T ~ F
  ))
  
gtex_sum_spec <- gtex_flip_spec %>%
  group_by(locus_name, Causal_SNP, Ref, Alt, Disease_expanded, variant_id, gtex_allele_match) %>%
  summarise(gtex_egenes_spec = paste(hgnc, collapse=";"),
            gtex_dir_agree_spec = case_when(
              sum(gtex_dir_flip < 0) == length(gtex_dir_flip) ~ "agree",
              sum(gtex_dir_flip > 0) == length(gtex_dir_flip) ~ "agree",
              T ~ "mixed"),
            gtex_dir_spec = paste(gtex_dir_flip, collapse=";"),
            gtex_tissue_spec = paste(tissue, collapse=";"),
            gtex_egenes_num_spec = length(unique(hgnc)))


res_merge_gtex_spec <- left_join(res_merge_gtex,
                          dplyr::select(ungroup(gtex_sum_spec), locus_name, variant_id,
                                        gtex_egenes_spec, gtex_tissue_spec, 
                                        gtex_dir_spec, gtex_dir_agree_spec, gtex_egenes_num_spec))
res_merge_gtex_spec <- res_merge_gtex_spec %>% 
  mutate(gtex_egenes_num_spec = ifelse(is.na(gtex_egenes_num_spec), 0, gtex_egenes_num_spec),
         gtex_dir_agree_spec = ifelse(is.na(gtex_dir_agree_spec), "NA", gtex_dir_agree_spec))

ggplot(res_merge_gtex_spec, aes(x = gtex_egenes_num_spec, fill =gtex_dir_agree_spec)) + 
  geom_histogram(bins=23) + 
  ggtitle("Number of GTEx e-genes per SNP (tissue-specific)") +
  xlab("Number of GTEx e-genes") +
  ylab("Number of SNPs") +
  labs(fill = "Do directions \nagree?") +
  scale_fill_viridis(discrete=TRUE, option ="D") +
  theme_classic() 

res_merge_gtex_hmec_hits_spec <- dplyr::filter(res_merge_gtex_spec, abs(hmec_logFC) > 0.5 & hmec_fdr < 0.01 & grepl("brca", Disease_expanded))

res_merge_gtex_hmec_hits_spec <- res_merge_gtex_hmec_hits_spec %>%  rowwise() %>%
  mutate(gtex_spec_hmecmpra_agree = case_when(
    gtex_egenes_num_spec == 0 ~ "NA",
    sum(unlist(str_split(gtex_dir_spec, ";")) < 0) > 0 & 
      sum(unlist(str_split(gtex_dir_spec, ";")) > 0) > 0 ~ "mixed",
    sum(unlist(str_split(gtex_dir_spec, ";")) < 0) == length(unlist(str_split(gtex_dir_spec, ";"))) &
    hmec_logFC < 0 ~ "down",
  sum(unlist(str_split(gtex_dir_spec, ";")) > 0) == length(unlist(str_split(gtex_dir_spec, ";"))) &
    hmec_logFC > 0 ~ "up",
  T ~ "disagree"))  

ggplot(res_merge_gtex_hmec_hits_spec, aes(x = gtex_egenes_num_spec, fill =gtex_spec_hmecmpra_agree)) + 
  geom_histogram(bins = 13) + 
  ggtitle("Number of GTEx e-genes per SNP, \n BRCA-specific HMEC hits only") +
  xlab("Number of GTEx e-genes") +
  ylab("Number of SNPs") +
  labs(fill = "Does GTEx \ndirection agree \nwith HMEC MPRA?") +
  scale_fill_viridis(discrete=TRUE, option ="D") +
  theme_classic() 

```

```{r}
write_tsv(res_merge_gtex_spec, "~/Documents/MPRA/Cancer_MPRA_Scaleup/eGeneFiles/eQTLgen/res_merge_with_eQTLGen_GTExreformat_11.1.20.tsv")
res_merge_gtex_spec <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/eGeneFiles/eQTLgen/res_merge_with_eQTLGen_GTExreformat_11.1.20.tsv")

```


How much overlap between egene methods?
Possibilities:
eqtl-gen
gtex_all
gtex_spec
hichip
eqtl-gen, gtex_all
eqtl-gen, gtex_spec
eqtl-gen, hichip
gtex_all, hichip
eqtl-gen, hichip, gtex_all



```{r fig.width=2.5, fig.height=2.5}

# first try and figure out how many shared between eqtl-gen and gtex_all
res_merge_overlap <- res_merge_gtex_spec %>% rowwise() %>%
  mutate(eqtlgen_gtexall_overlap =
           sum(unique(unlist(str_split(gtex_egenes_all, ";"))) %in%
                 unique(unlist(str_split(eqtlgen_egenes, ";")))))


res_merge_overlap <- res_merge_gtex_spec %>% rowwise() %>%
  mutate(overlap =
           case_when(
             !is.na(gtex_egenes_all) & !is.na(eqtlgen_egenes) & !is.na(HiChIP_egene) &
               length(c(intersect(intersect(unique(unlist(str_split(gtex_egenes_all, ";"))),
                     unique(unlist(str_split(eqtlgen_egenes, ";")))),
                     unique(unlist(str_split(HiChIP_egene, ";")))))) > 0 ~ "GTEx_eQTLGen_HiChIP",
             !is.na(gtex_egenes_all) & !is.na(eqtlgen_egenes) &
               length(intersect(unique(unlist(str_split(gtex_egenes_all, ";"))),
                     unique(unlist(str_split(eqtlgen_egenes, ";"))))) > 0 ~ "GTEx_eQTLGen",
             !is.na(gtex_egenes_all) & !is.na(HiChIP_egene) &
               length(intersect(unique(unlist(str_split(gtex_egenes_all, ";"))),
                     unique(unlist(str_split(HiChIP_egene, ";"))))) > 0 ~ "GTEx_HiChIP",
             !is.na(eqtlgen_egenes) & !is.na(HiChIP_egene) &
               length(intersect(unique(unlist(str_split(HiChIP_egene, ";"))),
                     unique(unlist(str_split(eqtlgen_egenes, ";"))))) > 0 ~ "eQTLGen_HiChIP",
             is.na(gtex_egenes_all) & is.na(eqtlgen_egenes) & is.na(HiChIP_egene) ~ "no_egene",
           T ~ "no_agreement"))
            
res_merge_overlap$overlap <- factor(res_merge_overlap$overlap,
                                       levels=c("GTEx_eQTLGen_HiChIP", "GTEx_HiChIP", 
                                                "eQTLGen_HiChIP", 
                                                "GTEx_eQTLGen", "no_agreement", "no_egene"), ordered=T)
ggplot(res_merge_overlap, aes(x=overlap)) + geom_bar() +
  ggtitle("Overlap between e-gene identification methods") + theme_classic() +
  theme(axis.text.x = element_text(angle = 90))

 
# find if a snp has a gold - eqtl-gen, hichip, gtex_all
res_merge_overlap_all <- res_merge_gtex_spec %>% rowwise() %>%
  mutate(eqtlgen_gtexall_hichip_overlap =
           case_when(
             is.na(gtex_egenes_all) | is.na(HiChIP_egene) | is.na(eqtlgen_egenes) ~ "",
             T ~ paste(c(intersect(intersect(unique(unlist(str_split(gtex_egenes_all, ";"))),
                     unique(unlist(str_split(eqtlgen_egenes, ";")))),
                     unique(unlist(str_split(HiChIP_egene, ";"))))), collapse=";")))
# about 77 genes
res_merge_overlap_sig <- res_merge_overlap %>% 
  dplyr::filter(case_when(
    grepl("airway", Disease_expanded) & airway_fdr < 0.01 & abs(airway_logFC) > 0.5 ~ T,
    grepl("ast", Disease_expanded) & ast_fdr < 0.01 & abs(ast_logFC) > 0.5 ~ T,
    grepl("brca", Disease_expanded) & hmec_fdr < 0.01 & abs(hmec_logFC) > 0.5 ~ T,
    grepl("colon", Disease_expanded) & colon_fdr < 0.01 & abs(colon_logFC) > 0.5 ~ T,
    grepl("endo", Disease_expanded) & endo_fdr < 0.01 ~ T & abs(endo_logFC) > 0.5,
    grepl("eso", Disease_expanded) & eso_fdr < 0.01 ~ T & abs(eso_logFC) > 0.5,
    grepl("lymphoma", Disease_expanded) & gm_fdr < 0.01 & abs(gm_logFC) > 0.5 ~ T,
    grepl("thy", Disease_expanded) & thy_fdr < 0.01 ~ T & abs(thy_logFC) > 0.5,
    grepl("pros", Disease_expanded) & pros_fdr < 0.01 ~ T & abs(pros_logFC) > 0.5,
    grepl("kc", Disease_expanded) & kc_fdr < 0.01 ~ T & abs(kc_logFC) > 0.5,
    grepl("melanoma", Disease_expanded) & mc_fdr < 0.01 & abs(mc_logFC) > 0.5 ~ T,
    grepl("panc", Disease_expanded) & panc_fdr < 0.01 & abs(panc_logFC) > 0.5 ~ T,
    grepl("renal", Disease_expanded) & renal_fdr < 0.01 & abs(renal_logFC) > 0.5 ~ T,
    grepl("ovca", Disease_expanded) & ov_fdr < 0.01 & abs(ov_logFC) > 0.5 ~ T))

  #   grepl("ast", Disease_expanded) & grepl("Brain", tissue) ~ T,
  #   grepl("brca", Disease_expanded) & grepl("Breast", tissue) ~ T,
  #   grepl("colon", Disease_expanded) & grepl("Colon", tissue) ~ T,
  #   grepl("endo", Disease_expanded) & grepl("Uterus", tissue) ~ T,
  #   grepl("eso", Disease_expanded) & grepl("Esophagus", tissue) ~ T,
  #   grepl("lymphoma", Disease_expanded) & grepl("EBV", tissue) ~ T,
  #   grepl("thy", Disease_expanded) & grepl("Thyroid", tissue) ~ T,
  #   grepl("pros", Disease_expanded) & grepl("Prostate", tissue) ~ T,
  #   grepl("kc", Disease_expanded) & grepl("Skin", tissue) ~ T,
  #   grepl("melanoma", Disease_expanded) & grepl("Skin", tissue) ~ T,
  #   grepl("panc", Disease_expanded) & grepl("Pancreas", tissue) ~ T,
  #   grepl("renal", Disease_expanded) & grepl("Kidney", tissue) ~ T,
  #   grepl("ovca", Disease_expanded) & grepl("Ovary", tissue) ~ T,
  #   T ~ F
  # ))
# check gtex_spec and eqtl-gen
res_merge_overlap <- res_merge_gtex_spec %>% rowwise() %>%
  mutate(eqtlgen_gtexall_overlap =
           sum(unique(unlist(str_split(gtex_egenes_all, ";"))) %in%
                 unique(unlist(str_split(eqtlgen_egenes, ";")))))


```






```{r}
# check on the alleles and whether directions have been flipped
sum(res_merge_eg$alt == res_merge_eg$Alt, na.rm=T)
# 4472
sum(!is.na(res_merge_eg$alt))
# 5019
# some alleles have mismatches, but we only care if they also have egenes
test <- res_merge_eg[!(res_merge_eg$Alt == res_merge_eg$alt) & !is.na(res_merge_eg$gtex_egenes),]
test <- dplyr::select(test, locus_name, Causal_SNP, Ref, Alt, Disease_expanded, alt, variant_id, gtex_egenes, 
                      gtex_egenes_correct_tissue, gtex_tissues, gtex_direction)
# 445 mismatches. 134 are deletions, where GTEx convention writes TG/T and MPRA writes G/-
sum(test$Ref == "-" | test$Alt == "-")
# 134
test <- dplyr::filter(test, !(Ref == "-" | Alt == "-"))
# 311
# these mostly seem like other alternate alleles or ambiguities. 
gtex_sum <- res_merge_eg %>% rowwise() %>% 
  mutate(num_gtex_alltissues = ifelse(is.na(gtex_egenes), 0, length(unlist(str_split(gtex_egenes, ",")))),
         num_gtex_spec =  ifelse(is.na(gtex_egenes_correct_tissue), 0,
                                 length(unlist(str_split(gtex_egenes_correct_tissue, ",")))))
gtex_sum <- gtex_sum %>% mutate(gtex_dir_agree = case_when(
  is.na(gtex_direction) ~ "NA",
  length(unlist(str_split(gtex_direction, ","))) == 1 ~ "agree",
  sum(unlist(str_split(gtex_direction, ",")) < 0) > 0 &
           sum(unlist(str_split(gtex_direction, ";")) > 0) > 0 ~ "mixed",
  T ~ "agree"))

```



```{r}
check_mpra_dir <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/Sequencingv3/NextSeq_Pool1/nextseq_pool1_counts_oldpGFcts.tsv")
check_mpra_dir_locus_2267 <- dplyr::filter(check_mpra_dir, grepl("locus_2267", oligo))
# ref higher, mpra logFC higher
```


Make an upsetr plot
```{r}
lt = 
m = make_comb_mat(lt)

```

