---
title: "Untitled"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(readr)
```
Read in motif breakr data from MG and MPRA res_merge
```{r}
motifs <- read_csv("~/Documents/MPRA/Cancer_MPRA_Scaleup/MotifBreakRAnalysis_MG/motif_df.csv")
res_merge <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/Analysis11.12.20/Analysis11.27.20/res_merge_withhitannot_031021.tsv")
# motif_clusters <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/MotifBreakRAnalysis_MG/RSAT_HOCOMOCOclustered_2.17.21.tab.txt", col_names = c("cluster", "motifs"))
hit_annot <- res_merge
```

Figure out which SNPs have a strongly disrupted motif
```{r}
motif_strong <- dplyr::filter(motifs, effect == "strong")
motif_strong <- motif_strong %>% group_by(SNP_id) %>%
  summarise(motif_effect = "strong")
motif_merge <- hit_annot %>% dplyr::select(Causal_SNP, hit_spec)
motif_merge <- left_join(motif_merge, motif_strong, by=c("Causal_SNP" = "SNP_id"))

motif_strong_weak <- dplyr::filter(motifs, effect %in% c("strong", "weak"))
motif_strong_weak <- motif_strong_weak %>% group_by(SNP_id) %>%
  summarise(motif_effect = "strong_or_weak")
motif_merge_all <- hit_annot %>% dplyr::select(Causal_SNP, hit_spec)
motif_merge_all <- left_join(motif_merge_all, motif_strong_weak, by=c("Causal_SNP" = "SNP_id"))

sum(motif_merge$motif_effect == "strong", na.rm=T)
# 2359 / 5400
# 43.7%
sum(motif_merge$motif_effect == "strong" & !is.na(motif_merge$hit_spec), na.rm=T)
# 200/421
# 47.5%
# 421 included duplicate snps, do out of 411 unique hits
# 200/411 = 48.7
sum(!is.na(motif_merge$hit_spec), na.rm=T)

```



```{r}
# reformat and collapse motif df
motif_coll_strong <- dplyr::filter(motifs, effect=="strong") %>% group_by(SNP_id) %>% 
  summarise(motifs = paste(unique(geneSymbol), collapse=","),
    motifs_gained = paste(unique(geneSymbol[allele_dir_bool == "gained"]), collapse=","),
            motifs_lost = paste(unique(geneSymbol[allele_dir_bool == "broken"]), collapse=","))
motif_coll_all <- motifs %>% group_by(SNP_id) %>% 
  summarise(motifs = paste(unique(geneSymbol), collapse=","),
    motifs_gained = paste(unique(geneSymbol[allele_dir_bool == "gained"]), collapse=","),
            motifs_lost = paste(unique(geneSymbol[allele_dir_bool == "broken"]), collapse=","))

motif_coll <- motif_coll_strong


hit_motif <- left_join(hit_annot, motif_coll, by=c("Causal_SNP"="SNP_id"))
hit_motif <- dplyr::filter(hit_motif, !is.na(disease))
# hit_motif <- hit_motif %>% mutate(motifs = str_split(motifs, ","))
```

Calculate enrichment and p values for hits, but first remove the ones that are 0, then multiple hypothesis correct the rest

```{r fig.height = 5, fig.width = 4}
hit_motif <- dplyr::select(hit_motif, Causal_SNP, disease, hit_spec, motifs, motifs_gained, motifs_lost)
# test all combos with more than 3 instances of the motif
combos_to_test <- dplyr::select(hit_motif, disease, motifs) %>%
  dplyr::filter(!is.na(motifs)) %>%
  mutate(motifs = str_split(motifs, ","), disease = str_split(disease, ",")) %>% 
  unnest(cols=c(motifs)) %>% unnest(cols=c(disease)) %>%
  group_by(disease, motifs) %>%
  summarise(keep = sum(!is.na(motifs)) > 3)
combos_to_test <- dplyr::filter(combos_to_test, keep ==T) %>% distinct()

calc_motif_per_disease_est <- function(motif, disease, df){
  # are motifs more likely to fall on hits?
  no_hit_motif = sum(grepl(motif, df$motifs) & grepl(disease, df$disease) & !(grepl(disease, df$hit_spec)))
  hit_in_motif = sum(grepl(motif, df$motifs) & grepl(disease, df$hit_spec))
  hit_no_motif = sum(!grepl(motif, df$motifs) & grepl(disease, df$hit_spec))
  no_hit_no_motif = sum(!grepl(motif, df$motifs) & grepl(disease, df$disease))
  mat = matrix(c(hit_in_motif, no_hit_motif, hit_no_motif, no_hit_no_motif),
               nrow=2)
  fish = fisher.test(mat)
  return(fish$estimate)
}



calc_motif_per_disease_pval <- function(motif, disease, df){
  # are motifs more likely to fall on hits?
  no_hit_motif = sum(grepl(motif, df$motifs) & grepl(disease, df$disease) & !(grepl(disease, df$hit_spec)))
  hit_in_motif = sum(grepl(motif, df$motifs) & grepl(disease, df$hit_spec))
  hit_no_motif = sum(!grepl(motif, df$motifs) & grepl(disease, df$hit_spec))
  no_hit_no_motif = sum(!grepl(motif, df$motifs) & grepl(disease, df$disease))
  mat = t(matrix(c(hit_in_motif, no_hit_motif, hit_no_motif, no_hit_no_motif),
               nrow=2))
  fish = fisher.test(mat)
  return(fish$p.value)
}

fisherest <- apply(combos_to_test, 1, function(x) calc_motif_per_disease_est(x["motifs"], x["disease"], hit_motif))


fisherp <- apply(combos_to_test, 1, function(x) calc_motif_per_disease_pval(x["motifs"], x["disease"], hit_motif))
combos_to_test$fisherp <- fisherp

fdr <- p.adjust(fisherp, method="BH")
combos_to_test$fdr <- fdr
diseases = unique(unlist(str_split(hit_motif$disease, ",")))
fisherest_df= sapply(diseases, 
                          function(y) sapply(unique(combos_to_test$motifs), 
                                             function(x) calc_motif_per_disease_est(x, y, hit_motif)))
fisherest_df[fisherest_df == Inf] <- max(fisherest_df[!fisherest_df == Inf])
fisherest_df_hm <- fisherest_df[!rowSums(fisherest_df) == 0,]
library(pheatmap)
pheatmap(t(fisherest_df_hm))
fisherest_df_hm <- fisherest_df_hm[str_remove(rownames(fisherest_df_hm), "\\..*") %in%
                                            combos_to_test$motifs[combos_to_test$fisherp < 0.05],]
fisherest_df_hm_log <- data.frame(fisherest_df_hm) %>% 
  mutate_all(function(x) case_when(x == 0 ~ 0,
                                   T ~ log2(x)))

rownames(fisherest_df_hm_log) <- str_remove(rownames(fisherest_df_hm), "\\..*")
pheatmap(t(fisherest_df_hm_log))
sig_combos <- combos_to_test[combos_to_test$fisherp < 0.1,]
fisherest_df_hm_log_cut <- fisherest_df_hm_log[rownames(fisherest_df_hm_log) %in% sig_combos$motifs,]

pheatmap(fisherest_df_hm_log_cut, main = "Log2 odds ratio of a disease-specific hit being in a motif")
colnames(fisherest_df_hm_log_cut)
fisherest_df_hm_log_cut_cols <- dplyr::select(fisherest_df_hm_log_cut, BRCA = hmec, THCA = thy,LUAD = airway,
                                              DLBC = gm, COAD = colon, PRAD = pros,
                                              SKCM = mc, UCEC = endo, OVCA = ov,
                                              KC = kc)
pheatmap(fisherest_df_hm_log_cut_cols, main = "Log2 odds ratio of a disease-specific hit being in a motif",
         fontsize_row = 9)

```



```{r}
cluster_fisherest= sapply(diseases, 
                          function(y) sapply(unique(hit_motif_cluster$cluster[!is.na(hit_motif_cluster$cluster)]), 
                                             function(x) calc_motif_cluster_est(x, y, hit_motif_cluster)))
cluster_fisherest[cluster_fisherest == Inf] <- 0
cluster_fisherest_hm <- cluster_fisherest[!rowSums(cluster_fisherest) == 0,]
cluster_fisherest_hm <- cluster_fisherest[str_remove(rownames(cluster_fisherest), "\\..*") %in%
                                            combos_cluster$cluster[combos_cluster$fisherp < 0.05],]
cluster_fisherest_hm_log <- data.frame(cluster_fisherest_hm) %>% 
  mutate_all(function(x) case_when(x == 0 ~ 0,
                                   T ~ log2(x)))
rownames(cluster_fisherest_hm_log) <- str_remove(rownames(cluster_fisherest_hm), "\\..*")
combos_cluster$fisherp <- fisherp_cluster
fdr_cluster <- p.adjust(fisherp_cluster, method="BH")
combos_cluster$fdr <- fdr_cluster


library(pheatmap)
pheatmap(cluster_fisherest_hm_log, main = "Log2 odds ratio of a disease-specific hit being in a motif")

```





Convert motifs to RSAT clusters - reduces motif number
```{r}
hit_motif_cluster <- hit_motif %>% mutate(motif = str_split(motifs, ",")) %>%
  unnest(cols =c(motif))
hit_motif_cluster <- hit_motif_cluster %>% rowwise() %>%
  mutate(cluster = ifelse(is.na(motif), NA,
                          ifelse(!sum(grepl(motif, motif_clusters$motifs)) == 1,  motif,
                                 motif_clusters$cluster[grepl(motif, motif_clusters$motifs)])))

combos_cluster <- dplyr::select(hit_motif_cluster, disease, cluster) %>%
  dplyr::filter(!is.na(cluster)) %>%
  mutate(cluster = str_split(cluster, ","), disease = str_split(disease, ",")) %>% 
  unnest(cols=c(cluster)) %>% unnest(cols=c(disease)) %>%
  group_by(disease, cluster) %>%
  summarise(keep = sum(!is.na(cluster)) > 1)
combos_cluster <- dplyr::filter(combos_cluster, keep ==T) %>% distinct()

calc_motif_cluster_pval <- function(motif, disease, df){
  # are motifs more likely to fall on hits?
  no_hit_motif = sum(grepl(motif, df$cluster) & grepl(disease, df$disease) & !(grepl(disease, df$hits)))
  hit_in_motif = sum(grepl(motif, df$cluster) & grepl(disease, df$hits))
  hit_no_motif = sum(!grepl(motif, df$cluster) & grepl(disease, df$hits))
  no_hit_no_motif = sum(!grepl(motif, df$cluster) & grepl(disease, df$disease))
  mat = t(matrix(c(hit_in_motif, no_hit_motif, hit_no_motif, no_hit_no_motif),
               nrow=2))
  fish = fisher.test(mat)
  return(fish$p.value)
}

calc_motif_cluster_est <- function(motif, disease, df){
  # are motifs more likely to fall on hits?
  no_hit_motif = sum(grepl(motif, df$cluster) & grepl(disease, df$disease) & !(grepl(disease, df$hits)))
  hit_in_motif = sum(grepl(motif, df$cluster) & grepl(disease, df$hits))
  hit_no_motif = sum(!grepl(motif, df$cluster) & grepl(disease, df$hits))
  no_hit_no_motif = sum(!grepl(motif, df$cluster) & grepl(disease, df$disease))
  mat = matrix(c(hit_in_motif, no_hit_motif, hit_no_motif, no_hit_no_motif),
               nrow=2)
  fish = fisher.test(mat)
  return(fish$estimate)
}

fisherp_cluster <- apply(combos_cluster, 1, function(x) calc_motif_cluster_pval(x["cluster"], x["disease"], hit_motif_cluster))
fisherest_cluster <- apply(combos_cluster, 1, function(x) calc_motif_cluster_est(x["cluster"], x["disease"], hit_motif_cluster))

cluster_fisherest= sapply(diseases, 
                          function(y) sapply(unique(hit_motif_cluster$cluster[!is.na(hit_motif_cluster$cluster)]), 
                                             function(x) calc_motif_cluster_est(x, y, hit_motif_cluster)))
cluster_fisherest[cluster_fisherest == Inf] <- 0
cluster_fisherest_hm <- cluster_fisherest[!rowSums(cluster_fisherest) == 0,]
cluster_fisherest_hm <- cluster_fisherest[str_remove(rownames(cluster_fisherest), "\\..*") %in%
                                            combos_cluster$cluster[combos_cluster$fisherp < 0.05],]
cluster_fisherest_hm_log <- data.frame(cluster_fisherest_hm) %>% 
  mutate_all(function(x) case_when(x == 0 ~ 0,
                                   T ~ log2(x)))
rownames(cluster_fisherest_hm_log) <- str_remove(rownames(cluster_fisherest_hm), "\\..*")
combos_cluster$fisherp <- fisherp_cluster
fdr_cluster <- p.adjust(fisherp_cluster, method="BH")
combos_cluster$fdr <- fdr_cluster


library(pheatmap)
pheatmap(cluster_fisherest_hm_log, main = "Log2 odds ratio of a disease-specific hit being in a motif")

```











```{r fig.width = 3, fig.height = 3}
hit_motif_res <- left_join(hit_annot, motif_coll, by=c("Causal_SNP"="SNP_id"))

motif_nondisspec_est <- function(motif, cell_type, df){
  # are motifs more likely to fall on hits?
  fdr_col <- df[grepl("fdr", colnames(df)) & grepl(cell_type, colnames(df))]
  no_hit_motif = sum(grepl(motif, df$motifs) & fdr_col[,1] > 0.05)
  hit_in_motif = sum(grepl(motif, df$motifs) & fdr_col[,1] < 0.05)
  hit_no_motif = sum(!grepl(motif, df$motifs) & fdr_col[,1] < 0.05)
  no_hit_no_motif = sum(!grepl(motif, df$motifs) & fdr_col[,1] > 0.05)
  mat = matrix(c(hit_in_motif, no_hit_motif, hit_no_motif, no_hit_no_motif),
               nrow=2)
  fish = fisher.test(mat)
  return(fish$estimate)
}

motif_nondisspec_pval <- function(motif, cell_type, df){
  # are motifs more likely to fall on hits?
  fdr_col <- df[grepl("fdr", colnames(df)) & grepl(cell_type, colnames(df))]
  no_hit_motif = sum(grepl(motif, df$motifs) & fdr_col[,1] > 0.05)
  hit_in_motif = sum(grepl(motif, df$motifs) & fdr_col[,1] < 0.05)
  hit_no_motif = sum(!grepl(motif, df$motifs) & fdr_col[,1] < 0.05)
  no_hit_no_motif = sum(!grepl(motif, df$motifs) & fdr_col[,1] > 0.05)
  mat = matrix(c(hit_in_motif, no_hit_motif, hit_no_motif, no_hit_no_motif),
               nrow=2)
  fish = fisher.test(mat)
  return(fish$p.value)
}

# motifs_present <- hit_motif %>% mutate(motif = str_split(motifs, ",")) %>%
#  unnest(cols= motif) %>% group_by(motif) %>% summarise(keep = length(Causal_SNP))


motif_enrich_nondisspec_est = sapply(cells, 
                          function(y) sapply(all_motifs, 
                                             function(x) motif_nondisspec_est(x, y, hit_motif_res)))
motif_enrich_nondisspec_p = sapply(cells, 
                          function(y) sapply(all_motifs, 
                                             function(x) motif_nondisspec_pval(x, y, hit_motif_res)))
test_fdr <- p.adjust(motif_enrich_nondisspec_p, method="BH")
enrich_nonspec_pivot <- rownames_to_column(data.frame(motif_enrich_nondisspec_p))
enrich_nonspec_pivot <- pivot_longer(enrich_nonspec_pivot, cols = colnames(enrich_nonspec_pivot)[-1], names_to = "cell_type", values_to = "pval")
enrich_nonspec_pivot$fdr <- p.adjust(enrich_nonspec_pivot$pval, method="BH")
motif_enrich_nondisspec_est[motif_enrich_nondisspec_est == Inf] <- 0
quantile(motif_enrich_nondisspec_est)
motif_enrich_nondisspec_p
nondisspec_hm <- motif_enrich_nondisspec_est[!rowSums(motif_enrich_nondisspec_est)== 0,]
nondisspec_hm <- motif_enrich_nondisspec_est[apply(motif_enrich_nondisspec_p, 1, function(x) sum(x < 0.05) > 0),]
test <- str_remove(rownames(motif_enrich_nondisspec_est), "\\..*") %in% 
                                               enrich_nonspec_pivot$rowname[enrich_nonspec_pivot$fdr < 0.1]

nondisspec_hm <- motif_enrich_nondisspec_est[str_remove(rownames(motif_enrich_nondisspec_est), "\\..*") %in% 
                                               enrich_nonspec_pivot$rowname[enrich_nonspec_pivot$fdr < 0.1] &
                                               !(str_remove(rownames(motif_enrich_nondisspec_est), "\\..*") == "T"),]
nondisspec_hm_log <- data.frame(nondisspec_hm) %>% 
  mutate_all(function(x) case_when(x == 0 ~ 0,
                                   T ~ log2(x)))
rownames(nondisspec_hm_log) <- str_remove(rownames(nondisspec_hm), "\\..*")
pheatmap(nondisspec_hm_log, fontsize_row = 8, main = "Log2 odds ratio of an active sequence being within a motif\n(not disease specific)")
         



```

```{r fig.width = 7, fig.height = 3}
qmpra <- read_tsv("~/Documents/MPRA/Cancer_MPRA_Scaleup/QuantitativeMPRAnalyze/alpha_results_1.28.21.tsv")
qmpra <- dplyr::filter(qmpra, grepl("motif", rowname))
qmpra_hm <- qmpra %>% mutate(rowname = str_remove(str_remove(rowname, "^motif_"), "_motif$"))
qmpra_hm <- qmpra_hm %>% column_to_rownames("rowname")
qmpra_hm[is.na(qmpra_hm)] <- 0
q_sds <- apply(qmpra_hm, 1, sd)
quantile(q_sds, seq(0, 1, 0.1), na.rm=T)
test <- qmpra_hm[q_sds > quantile(q_sds,  seq(0, 1, 0.1), na.rm=T)[10],]
pheatmap(t(qmpra_hm[q_sds > quantile(q_sds, seq(0, 1, 0.1), na.rm=T)[10],]), fontsize_col = 8,
         main = "Transcription level of different motifs across cell types")
         
```

